free-1.md

```diff
@@ -4,65 +4,176 @@
+how we might naturally think about our schedule.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+The data supplied by the user (either through an intuitive language or an
+exported document, or some amalgamation of the two) would be transformed to
+the abstract representation of a schedule.
+
+If there are multiple abstract representations, different operations
+may be performed upon them (decided by the organizer), such as intersection,
+greedy algorithms for free individuals, multiple meeting times that cover
+all individuals etc.  (The language should provide a way for users to provide
+a custom combination functions from modular pieces.)
+
+It should also provide a color-coded graphical picture of the schedule
+information the user input, so they can verify that the information they
+provided was parsed correctly (and entered correctly).  This would be the
+primary mechanism for reporting semantic errors.  Syntax and failure to
+parse errors would be reported on a best guess line number basis.
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to describe when are the great, good, bad, and terrible
+times to meet (likely default values).  It should also be possible to
+express more complicated dependencies, such as a preference for a meeting
+time that coincides or avoids another group member, or a freedom in a 3
+hour block where only 1 block can be scheduled at a time.
+
+It should be very difficult to do perform functions unrelated to scheduling,
+such as arithmetic, nested control flow, or see implementation details.
+
+I don't think there should be anything in between (of medium difficulty).
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+We might consider the aforementioned websites 
+[WhenIsGood](http://whenisgood.net/) and
+[When2Meet](http://www.when2meet.com), since they are computerized schedulers
+with a specific syntax (painting over certain blocks), and associated
+semantics (painted blocks are free blocks).
+
+These DSLs are quite minimalistic (i.e. when they run, they only support the
+intersection operation for the sets of times from each individual -- as far
+as I can tell).  My solution to the scheduling problem might be a bit more
+ambitious in the functionality it hopes to provide, but I will need to take
+care during design to avoid feature creep and the complexity for users that
+may entail.  Perhaps I will find that the existing websites' simplicity is
```

Comment:
I doubt it, personally, but I think that the added expressiveness is worthwhile. This could also be an interesting space for tooling - provide an interface where they can paint over blocks, and have it generate the code necessary.